packages/manager/src/features/Firewalls/FirewallLanding/CreateFirewallDrawer.tsx

```diff
@@ -356,49 +317,55 @@ export const CreateFirewallDrawer = React.memo(
-            errorText={errors['devices.linodes']}
-            helperText={deviceSelectGuidance}
-            multiple
-            optionsFilter={linodeOptionsFilter}
-            value={values.devices?.linodes ?? null}
+          <Controller
+            render={({ field, fieldState }) => (
+              <LinodeSelect
+                label={
+                  createFlow === 'linode' ? LINODE_CREATE_FLOW_TEXT : 'Linodes'
+                }
+                onSelectionChange={(linodes) => {
+                  field.onChange(linodes.map((linode) => linode.id));
+                }}
+                errorText={fieldState.error?.message}
+                helperText={deviceSelectGuidance}
+                multiple
+                optionsFilter={linodeOptionsFilter}
+                value={field.value ?? null}
+              />
+            )}
+            control={control}
+            name="devices.linodes"
           />
-          <NodeBalancerSelect
-            label={
-              createFlow === 'nodebalancer'
-                ? NODEBALANCER_CREATE_FLOW_TEXT
-                : 'NodeBalancers'
-            }
-            onSelectionChange={(nodebalancers) => {
-              setFieldValue(
-                'devices.nodebalancers',
-                nodebalancers.map((nodebalancer) => nodebalancer.id)
-              );
-            }}
-            // @ts-expect-error this form intentionally breaks Formik's error type
-            errorText={errors['devices.nodebalancers']}
-            helperText={deviceSelectGuidance}
-            multiple
-            optionsFilter={nodebalancerOptionsFilter}
-            value={values.devices?.nodebalancers ?? null}
+          <Controller
+            render={({ field, fieldState }) => (
+              <NodeBalancerSelect
+                label={
+                  createFlow === 'nodebalancer'
+                    ? NODEBALANCER_CREATE_FLOW_TEXT
+                    : 'NodeBalancers'
+                }
+                onSelectionChange={(nodebalancers) => {
+                  field.onChange(
+                    nodebalancers.map((nodebalancer) => nodebalancer.id)
+                  );
+                }}
+                errorText={fieldState.error?.message}
+                helperText={deviceSelectGuidance}
+                multiple
+                optionsFilter={nodebalancerOptionsFilter}
+                value={field.value ?? null}
+              />
+            )}
+            control={control}
+            name="devices.nodebalancers"
           />
           <ActionsPanel
             primaryButtonProps={{
               'data-testid': 'submit',
-              disabled: userCannotAddFirewall,
+              disabled: userCannotAddFirewall || !isDirty,
               label: 'Create Firewall',
               loading: isSubmitting,
-              type: 'submit',
+              onClick: handleSubmit(createCustomFirewall),
```

Comment:
Is there any reason we can't use the `type: 'submit'` pattern here?

```suggestion
              type: 'submit',
```

Reply:
putting back in `type: 'submit'` - rn it looks like I have to keep onClick, otherwise if I'm in the Linode Create form, submitting the firewall form would also try to submit the Linode Create form. I'd tried looking into it earlier but didn't get too far, will look into it some more